src/app/site/checkout/checkout-page/checkout-page.component.ts

```diff
@@ -85,28 +95,86 @@ export class CheckoutPageComponent implements OnInit {
       });
 
     this.checkoutService.isValid$.subscribe(r => this.disabled = (!r && !this.soldOuts));
+    if (!this.authService.userDetails.userId) {
+      this.showEarnPointLabel = false;
+      this.earnedLoyaltyPoint = 0;
+    } else {
+      this.showEarnPointLabel = true;
+      this.checkoutService.loyaltyGroups.subscribe(data => this.loyaltyGroups = data);
+      this.checkoutService.addPointArray.subscribe(data => this.addPointArray = data[0].add_point);
+    }
+
+    // this.checkoutService.setPaymentType(this.paymentType.cash);
+    this.checkoutService.setPaymentType(this.paymentType[this.selectedPaymentType]);
   }
 
   changePaymentType(data) {
     this.usedBalance = 0;
     this.usedLoyaltyPoint = 0;
+    this.selectedPaymentType = data;
```

Comment:
you set this property here and in setPaymentType function.
recommended to set this in related function if this function can be change (e.g. need some logic to change/add) or use this logic not only here (use it in some other parts)
or remove function and use this only here